lib/defql/macros/defselect.ex

```diff
@@ -4,27 +4,29 @@ defmodule Defql.Macros.Defselect do
   alias Defql.Connection
 
   @doc false
-  defmacro defselect({name, _, params}, [table: table]) when is_atom(name) and is_list(params) and length(params) == 1 and is_atom(table) do
+  defmacro defselect(_name_params, _opts \\ [])
+  @doc false
+  defmacro defselect({name, _, params}, [columns: columns] = opts) when is_atom(name) and is_list(params) and length(params) == 1 and is_list(columns) do
     [first | _] = params
     quote do
       def unquote(name)(unquote_splicing(params)) do
         Connection.select(
-                            unquote(table),
-                            unquote(first)
+                            resolve_table(unquote(opts)),
+                            unquote(first),
+                            unquote(columns)
                           )
       end
     end
   end
 
   @doc false
-  defmacro defselect({name, _, params}, [table: table, columns: columns]) when is_atom(name) and is_list(params) and length(params) == 1 and is_atom(table) and is_list(columns) do
+  defmacro defselect({name, _, params}, opts) when is_atom(name) and is_list(params) and length(params) == 1 do
     [first | _] = params
     quote do
       def unquote(name)(unquote_splicing(params)) do
         Connection.select(
-                            unquote(table),
-                            unquote(first),
-                            unquote(columns)
+                            resolve_table(unquote(opts)),
+                            unquote(first)
```

Comment:
sorry) my comments were `pending` I didn't get used to new review system yet) (didn't click 'finish review')